medusa/server/api/v1/core.py

```diff
@@ -1303,9 +1326,14 @@ def run(self):
         if not self.type:
             self.type = 'manual'
 
-        data = process_tv.ProcessResult(self.path, process_method=self.process_method).process(
-            force=self.force_replace, is_priority=self.is_priority, delete_on=self.delete_files,
-            failed=self.failed, proc_type=self.type
+        data = app.post_processor_scheduler.action.run(
```

Comment:
Because your directly calling the run method here, it's not triggered from the scheduler. The scheduler itself is running in a thread. But by calling action.run() your bypassing that.

The periodic scheduled pp, can be called by the the scheduler. But pp called from apiv1/nzbToMedia needs to be done in sync. So that will cause a block on the webserver thread.

Reply:
Thanks @p0psicles, @sharkykh and I already had a discussion about that and it is clear now. I will think about what to do tomorrow, because I don't want to add a Queue for this. The PR still fixes issues like #6799 and a bunch of other minor stuff.

Reply:
We can reserve a real async pp solution for when we move the pp API call to apiv2. Then we can also destinct the solution in nzbToMedia. And have it poll for the pp outcome.